client/azure_client_set.go

```diff
@@ -11,12 +11,194 @@ import (
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/adal"
 	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/giantswarm/exporterkit/collector"
 	"github.com/giantswarm/microerror"
+	"github.com/prometheus/client_golang/prometheus"
 
 	"github.com/giantswarm/azure-operator/client/senddecorator"
 	"github.com/giantswarm/azure-operator/pkg/backpressure"
 )
 
+var (
+	// Prometheus metrics for clients
+	deploymentsClientMetricsDesc = prometheus.NewSummaryVec(
```

Comment:
noob question: are there advantages to use different metrics instead of a single metric with different labels? Like having a single metric but using the client name as label?

Reply:
I'm not fully sure. Potentially both could work. Metric per client just made logically sense.

Reply:
I would have a label here, per client. Makes the alerting easier.

Reply:
I would use a label here, makes the alert simpler.

Reply:
Ack